cvmfs/options.h

```diff
@@ -15,24 +15,158 @@
+   * @return true if that variable was previously defined in
+   *         at least one source
+   */
+  bool GetSource(const std::string &key, std::string *value);
+
+  /**
+   * Checks if a variable contains a boolean value
+   *
+   * @param   param_value variable to be accessed in the map
+   * @return  true if param has as value "YES", "ON" or "1". False otherwise.
+   */
+  bool IsOn(const std::string &param_value);
+
+  /**
+   * Retrieves a vector containing all stored keys.
+   *
+   * @return a vector with all keys contained in the map
+   */
+  std::vector<std::string> GetAllKeys();
 
-void Init();
-void Fini();
+  /**
+   * Gets all stored key-values of the map in an string format. This format
+   * follows the following pattern:
+   *
+   * "KEY=VALUE    # from SOURCE"
+   *
+   * @return a vector containing all key-values in a string format.
+   */
+  std::string Dump();
 
-void ParsePath(const std::string &config_file, const bool external);
-void ParseDefault(const std::string &fqrn);
-void ClearConfig();
-bool IsDefined(const std::string &key);
-bool GetValue(const std::string &key, std::string *value);
-bool GetSource(const std::string &key, std::string *value);
-bool IsOn(const std::string &param_value);
-std::vector<std::string> GetAllKeys();
-std::string Dump();
+  /**
+   * Parse a configuration file whose variables's values are positive integers
+   *
+   * @param  path absolute path to the configuration file
+   * @param  map map to store the generated key-value
+   * @return true if the file exists, is readable and all values can be parsed
+   *  to integers
+   */
+  bool ParseUIntMap(const std::string &path, std::map<uint64_t, uint64_t> *map);
+
+ protected:
+  bool HasConfigRepository(const std::string &fqrn, std::string *config_path);
+
+ protected:
+  std::map<std::string, ConfigValue> config_;
+};  // class OptionManager
+
+
+/**
+ * Derived class from OptionsManager. This class provides a so-called fast
+ * parsing procedure which does not create a fork of the process and only
+ * retrieves those options of the configuration file which strictly are in the
+ * following format:
+ *
+ *  "KEY=VALUE".
+ *
+ *  No comments (#) are allowed.
+ *
+ *  @note In order to use this parse it is necessary to execute the program
+ *  with the "-o simple_options_parsing" flag
+ *  @note If using IgProf profiling tool it is necessary to use this parser in
+ *  order to avoid a fork.
+ */
+class FastOptionsManager : public OptionsManager {
+ public:
+  void ParsePath(const std::string &config_file, const bool external);
+};  // class FastOptionManager
+
+
+/**
+ * Derived class from OptionsManager. This class provides the original an most
+ * complete parsing of the configuration files. In order to parse the
+ * configuration files it retrieves the "KEY=VALUE" pairs and uses bash for
+ * the rest, so that you can execute sightly complex scripts.
+ *
+ * @note if no "-o simple_options_parsing" flag is passed to the program this
```

Comment:
I think this one and the next note can be removed, this already documented as part of the FastOptionsManager